src/Surfnet/StepupSelfService/SelfServiceBundle/Service/RemoteVetting/Value/AttributeMatchCollection.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * Copyright 2020 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupSelfService\SelfServiceBundle\Service\RemoteVetting\Value;
+
+use ArrayIterator;
+use ArrayAccess;
+use IteratorAggregate;
+use JsonSerializable;
+
+class AttributeMatchCollection implements JsonSerializable, IteratorAggregate, ArrayAccess
+{
+    /**
+     * @var AttributeMatch[]
+     */
+    private $matches = [];
+
+    public static function fromAttributeCollection(AttributeCollection $attributeCollection)
```

Comment:
Incorporating the Saml Bundle AttributeDictionary to be able to match SurfConext attributes with RV IdP attributes might be a good idea. But not for now.

Reply:
The matching should later on be implemented in the `\Surfnet\StepupSelfService\SelfServiceBundle\Service\RemoteVettingService::getValidatingAttributes` method. So from that point on internally only the Collection is used. This method uses the AttruibuteSet from the bundle. If something changes later on, this only has to be changed in one spot.
Personally I don't like the use of the SamlBundle, the use of bundles for this kind of shared logic is also going against the Symfony best-practices, but that is a not a discussion for now.